### DISCUSSION TITLE
Add a GUI to QCoDeS (was make Qcodes more user-friendly)

### DISCUSSION THREAD

**Discussion by zaizhe** [microsoft/Qcodes#4444] [Ideas]
Hi, thank you very much for your Qcodes library that allows me to control my experimental equipment. We used to use the Labber software(https://www.keysight.com/us/en/products/software/application-sw/labber-software.html). This software has many functions, but the annoying thing is that this software is charged and can only be used for free for 30 days, I want to ask if the Qcodes library can provide some user interface, so that those of us who don't know python can get started more easily, using the GUI interface instead of jupyter-lab, I look forward to your Reply, thanks!

**Comment by jenshnielsen** [comment]
@zaizhe thanks for the question. I took the liberty of converting this to a Discussion. We use github discussions for more open ended questions like this.  I also changed the title a bit to make it more specifically reflect gui

**Comment by jenshnielsen** [comment]
Thanks for the question @zaizhe 

A gui is not really on our roadmap. We have found that the time required to build a reasonable gui does not really match the amount of time we would be able to invest in such a project. That being said: 

* We are absolutely welcoming anyone who would like to build a gui on top of QCoDeS and are aware of a number of projects that build on top of QCoDeS. For example [Plottr ](https://github.com/toolsforexperiments/plottr) which provides a gui for data exploration. 
* We are always working on making QCoDeS easier to use as a python library. We maintain a hopefully friendly tutorial [here ](https://qcodes.github.io/Qcodes/examples/15_minutes_to_QCoDeS.html) and are continuously working on making the code easier to use and better documented. Any feedback on improvements is much appreciated.
